build.spec

```diff
@@ -50,9 +51,16 @@ hiddenimports: list[str] = [
 ]
 
 if sys.platform == "linux":
-    # Needed files for better system tray support on Linux via pystray (AppIndicator backend).
-    datas.append((Path("/usr/lib/girepository-1.0/AppIndicator3-0.1.typelib"), "gi_typelibs"))
-    binaries.append((Path("/lib/x86_64-linux-gnu/libappindicator3.so.1"), "."))
+    arch = platform.machine()
+    if arch == "x86_64":
+        datas.append((Path("/usr/lib/x86_64-linux-gnu/girepository-1.0/AyatanaAppIndicator3-0.1.typelib"), "gi_typelibs"))
```

Comment:
Any strong reason for the switch to Ayatana's AppIndicator here?

Reply:
I couldn't get libappindicator3 to work on my Raspberry pi 5 (Raspberry os), and from the research i could find about it does it seem like an issue overall that it, like you aren't able to install the package at all etc (from what it looks like is gir1.2 outdated), the only solution seems to be using Ayatana's AppIndicator, sorry for not adding that in the pr at first

It also seems like it might be what this issue is about https://github.com/DevilXD/TwitchDropsMiner/issues/221

Maybe there's a better way to do it tbh, but it's the only solution i could fine that seems to have fixed it for me

Reply:
It's strange that the regular AppIndicator doesn't work for you. Do you get any errors in the terminal, or something? What version of Raspberry Pi OS are you running?

>It also seems like it might be what this issue is about https://github.com/DevilXD/TwitchDropsMiner/issues/221

That's actually documentation issue, but [this comment](https://github.com/DevilXD/TwitchDropsMiner/issues/221#issuecomment-1591610740) explains what are the required dependencies.

Reply:
yes! 

When i try to run the `sudo apt install gir1.2-appindicator3-0.1 libcairo2-dev libgirepository1.0-dev pkg-config` command:
```sh
0.1 libcairo2-dev libgirepository1.0-dev pkg-config
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package gir1.2-appindicator3-0.1 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'gir1.2-appindicator3-0.1' has no installation candidate
```

When i try to run the build:
```sh
Running pyinstaller...
159 INFO: PyInstaller: 6.11.1, contrib hooks: 2024.10
159 INFO: Python: 3.11.2
160 INFO: Platform: Linux-6.6.51+rpt-rpi-2712-aarch64-with-glibc2.36
160 INFO: Python environment: /home/tobia/Desktop/TwitchDropsMiner/env
178 INFO: Module search paths (PYTHONPATH):
['/home/tobia/Desktop/TwitchDropsMiner',
 '/usr/lib/python311.zip',
 '/usr/lib/python3.11',
 '/usr/lib/python3.11/lib-dynload',
 '/home/tobia/Desktop/TwitchDropsMiner/env/lib/python3.11/site-packages',
 '/home/tobia/Desktop/TwitchDropsMiner']
520 INFO: Appending 'binaries' from .spec
521 INFO: Appending 'datas' from .spec
Unable to find '/usr/lib/aarch64-linux-gnu/girepository-1.0/AppIndicator3-0.1.typelib' when adding binary and data files.
PyInstaller build failed.
```
(this fails obviously because appindicator3 isn't installed)

With the issue did i refer to some of the later comments, sorry should have been more clear about that

Honestly i agree, the issue is just i can't find any other way other than installing it manually i guess

Did i bit more digging and yeah it looks like AppIndicator3 is deprecated which is why i can't install it, so it might be a good idea to move to AyatanaAppIndicator3 either way
https://github.com/Ulauncher/Ulauncher/issues/702#issuecomment-803027651
https://www.reddit.com/r/debian/comments/yqqa27/comment/ivpn05b/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

Reply:
Oh, so AppIndicator-related packages got dropped in recent versions of Debian/Ubuntu... Gotcha. We should definitely switch to Ayatana then.

Reply:
Yeah at least that's what it looks like and that would explain why i can't install it too

Just to be 100% sure as you definitely know more than me on this, is there more than needs to be changed other than the thing in the workflow and in [AppImageBuilder.yml](https://github.com/DevilXD/TwitchDropsMiner/pull/617#discussion_r1842930395)? (they should be fixed now)

It works for me but idk if there's something missing within the actual scripts

Reply:
Everything looks clear now? 👀

Reply:
Linux-wise, yeah. I just want one minor nitpick.